src/changes.rs

```diff
@@ -166,15 +188,30 @@ impl<'a> ChangeSet<'a> {
             }
             WriteMode::NewFile(extn) => {
                 let filename = filename.to_owned() + "." + extn;
-                let mut file = try!(File::create(&filename));
-                try!(write!(file, "{}", text));
+                let file = try!(File::create(&filename));
+                try!(style(file, text));
             }
             WriteMode::Display => {
                 println!("{}:\n", filename);
-                println!("{}", text);
+                let lock = stdout();
+                let lock = lock.lock();
+                try!(style(lock, text));
             }
             WriteMode::Return(_) => {
-                return Ok(Some(text.to_string()));
+                return Ok(Some(match NEWLINE_STYLE {
+                    NewlineStyle::Unix => text.to_string(),
+                    NewlineStyle::Windows => {
+                        let mut s = String::new();
+                        for (c, _) in text.chars() {
+                            match c {
+                                '\n' => s.extend("\r\n".chars()),
+                                '\r' => continue,
+                                c => s.push(c),
+                            }
+                        }
+                        s
```

Comment:
Ah. the issue is `io::Write` vs `fmt::Write`. `String` only implements `fmt::Write`, but no other types implement `fmt::Write`. So it's not possible to do this with generics. Either I use a macro, or I do an additional `from_utf8`. I'll go with the latter version for now.